solr/core/src/java/org/apache/solr/security/RuleBasedAuthorizationPluginBase.java

```diff
@@ -110,6 +113,19 @@ public AuthorizationResponse authorize(AuthorizationContext context) {
     return flag.rsp;
   }
 
+  /**
+   * Retrieves permission names for a given set of roles
+   */
+  public Set<String> getPermissionNamesForRoles(Set<String> roles) {
+    Set<String> permissions = new HashSet<>();
+    for (String r : roles) {
+      if (roleToPermissionsMap.containsKey(r)) {
```

Comment:
Let’s use a fully functional approach with flat map rather than mixing here

Reply:
Done. Also added a unit test for this new method.

Reply:
The unit test uncovered a NPE that I also fixed.